near-sdk-macros/src/core_impl/info_extractor/impl_item_method_info.rs

```diff
@@ -62,18 +62,6 @@ mod tests {
         assert!(matches!(actual, ReturnType::Type(_, ty) if ty.as_ref() == &expected));
     }
 
-    #[test]
-    fn handle_result_incorrect_return_type() {
```

Comment:
I think we should put this back, see https://github.com/near/near-sdk-rs/pull/1057/files#r1272593025

Reply:
The whole point of this PR is to remove this check at the macro level because it's too restrictive. It prevents users from using type aliases and such (like `anyhow::Result<T>`).